tests/integration/block_createBlock_k8s_test.go

```diff
@@ -156,49 +161,54 @@ func (s *K8sBlockImageCreateSuite) TestCreatePVCWhenStorageClassExists() {
-	//create pool and storageclass
-	result1, err1 := s.storageClassOperation("test-pool-1", "create")
-	require.Contains(s.T(), result1, "pool \"test-pool-1\" created", "Make sure test pool is created")
-	require.Contains(s.T(), result1, "storageclass \"rook-block\" created", "Make sure storageclass is created")
+	logger.Infof("Test creating PVC(create block images) twice")
+	claimName := "test-twice-claim"
+	poolName := "test-twice-pool"
+	defer s.tearDownTest(claimName, poolName)
+	status, _ := s.kh.GetPVCStatus(claimName)
+	logger.Infof("PVC %s status: %s", claimName, status)
+	s.bc.BlockList()
+
+	logger.Infof("create pool and storageclass")
+	result1, err1 := s.storageClassOperation(poolName, "create")
+	assert.Contains(s.T(), result1, fmt.Sprintf("pool \"%s\" created", poolName), "Make sure test pool is created. "+result1)
+	assert.Contains(s.T(), result1, "storageclass \"rook-block\" created", "Make sure storageclass is created")
 	require.NoError(s.T(), err1)
 
-	//make sure storageclass is created
+	logger.Infof("make sure storageclass is created")
 	present, err := s.kh.IsStorageClassPresent("rook-block")
 	require.Nil(s.T(), err)
 	require.True(s.T(), present, "Make sure storageclass is present")
 
-	//create pvc
+	logger.Infof("create pvc")
 	result2, err2 := s.pvcOperation(claimName, "create")
-	require.Contains(s.T(), result2, "persistentvolumeclaim \"test-claim1\" created", "Make sure pvc is created")
+	require.Contains(s.T(), result2, fmt.Sprintf("persistentvolumeclaim \"%s\" created", claimName), "Make sure pvc is created. "+result2)
 	require.NoError(s.T(), err2)
 
-	//check status of PVC
+	logger.Infof("check status of PVC")
 	require.True(s.T(), s.isPVCBound(claimName))
 
-	b1, _ := s.bc.BlockList()
-	require.Equal(s.T(), s.initBlockCount+1, len(b1), "Make sure new block image is created")
+	b1, err := s.bc.BlockList()
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), s.initBlockCount+1, len(b1), "Make sure new block image is created")
 
-	//Create same pvc again
+	logger.Infof("Create same pvc again")
 	result3, err3 := s.pvcOperation(claimName, "create")
-	require.Contains(s.T(), result3, "persistentvolumeclaims \"test-claim1\" already exists", "make sure PVC is not created again")
+	require.Contains(s.T(), result3, fmt.Sprintf("persistentvolumeclaims \"%s\" already exists", claimName), "make sure PVC is not created again. "+result3)
 	require.NoError(s.T(), err3)
 
-	//check status of PVC
+	logger.Infof("check status of PVC")
 	require.True(s.T(), s.isPVCBound(claimName))
 
-	//check bock image count
+	logger.Infof("check block image count")
 	b2, _ := s.bc.BlockList()
-	require.Equal(s.T(), len(b1), len(b2), "Make sure new block image is created")
+	assert.Equal(s.T(), len(b1), len(b2), "Make sure new block image is created")
 
 }
 
-func (s *K8sBlockImageCreateSuite) TearDownTest() {
-
+func (s *K8sBlockImageCreateSuite) tearDownTest(claimName, poolName string) {
 	s.pvcOperation(claimName, "delete")
```

Comment:
how do we know the real pool got deleted? Can we use the pool api to get a list of pools to ensure that the pools got deleted? The test framework has a function to queries pools api

Reply:
yes, there are place the tests check for existence of pools by calling `BlockList()`

Reply:
Is there a test where the crd is deleted and then we do a listpools via the api to ensure true underlying pool is also deleted?

Reply:
True=the. I shouldn't be reviewing from a phone :)

Reply:
i didn't even notice the typo until you mentioned it. :) i'll add another test for this

Reply:
actually, there is already an e2e test, but i do need to add some more unit tests.
https://github.com/rook/rook/blob/master/tests/integration/base_block_test.go#L97